core/arch/arm/kernel/generic_boot.c

```diff
@@ -662,3 +661,30 @@ void generic_boot_init_secondary(unsigned long nsec_entry)
 	init_secondary_helper(nsec_entry);
 }
 #endif
+
+int generic_boot_core_release(size_t core_idx, paddr_t entry)
```

Comment:
As you pointed out, this function shall be protected by CFG_BOOT_SECONDARY_REQUEST, it is not used in build, so no need to do panic on the fly.